Client/Ecosia/UI/NTP/ClimateImpactCounter/ArchProgressView.swift

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+
+struct RectangularArcShape: Shape {
+    var progress: CGFloat // Progress value between 0 and 1
+    
+    func path(in rect: CGRect) -> Path {
+        var path = Path()
+        
+        // Calculate the start and end angles for the arc
+        let startAngle = Angle(degrees: 180)
+        let endAngle = Angle(degrees: 180 + Double(180 * progress))
+        
+        // Create an arc based on the rectangle's dimensions
+        path.addArc(center: CGPoint(x: rect.midX, y: rect.maxY),  // Center at the bottom of the rect
+                    radius: min(rect.width, rect.height) / 1.5,     // Use the smaller dimension for the radius
+                    startAngle: startAngle,
+                    endAngle: endAngle,
+                    clockwise: false)
+        
+        return path
+    }
+}
+
+struct ArchProgressView: View {
+    
+    var progress: CGFloat
+    var lineWidth: CGFloat = 10
+    @ObservedObject var theme: SeedTheme
```

Comment:
Nice usage of `ObservedObject` 👏 

Wouldn't it make more sense to call it `ArcTheme` though and leave that class inside this one instead of belonging to the `SeedCounterView`? That in my head makes more sense and makes it clearer it's reusable (since the way you built it is very reusable already ❤️)